app/src/db/migrations/20220130133615_000-init.js

```diff
@@ -0,0 +1,129 @@
+const stamps = require('../stamps');
+const uuid = require('uuid');
+
+const CREATED_BY = uuid.NIL;
+
+exports.up = function (knex) {
+  return Promise.resolve()
+
+    // Create user table and add a dummy system user (to link migration CreatedBy stamps)
+    .then(() => knex.schema.createTable('oidc_user', table => {
+      table.string('oidcId').primary();
+      table.string('firstName');
+      table.string('fullName');
+      table.string('lastName');
+      table.string('username').notNullable().index();
+      table.string('email').index();
+      table.boolean('active').notNullable().defaultTo(true);
+      stamps(knex, table);
+    }))
+    .then(() => {
+      const items = [
+        {
+          oidcId: CREATED_BY,
+          username: 'System',
+          active: false
+        },
+      ];
+      return knex('oidc_user').insert(items);
+    })
+    .then(() => knex.schema.createTable('identity_provider', table => {
+      table.string('code').primary();
+      table.string('display').notNullable();
+      table.string('idpAlias');
+      table.boolean('active').notNullable().defaultTo(true);
+      stamps(knex, table);
+    }))
+    // add the idp fk for user (user needs to be created first for stamp fks)
+    .then(() => knex.schema.alterTable('oidc_user', table => {
+      table.string('idp').references('code').inTable('identity_provider').after('oidcId');
+    }))
+
+
+    // Add the rest of the tables
+    .then(() => knex.schema.createTable('permission', table => {
+      table.string('code').primary();
+      table.string('display').notNullable();
+      table.boolean('active').notNullable().defaultTo(true);
+      stamps(knex, table);
+    }))
+    .then(() => knex.schema.createTable('object', table => {
+      table.uuid('id').primary();
+      table.string('originalName', 1024).notNullable();
+      table.string('path', 1024).notNullable();
+      table.string('mimeType').notNullable();
+      table.string('uploaderOidcId').references('oidcId').inTable('oidc_user');
```

Comment:
this is good for auditting. 
I guess an object will always be linked to a user.